src/routers/alpha-router/quoters/v2-quoter.ts

```diff
@@ -162,8 +170,10 @@ export class V2Quoter extends BaseQuoter<V2CandidatePools, V2Route> {
       gasPriceWei,
       poolProvider: this.v2PoolProvider,
       token: quoteToken,
-      providerConfig: _routingConfig,
-      // TODO: implement wrap overhead for v2 routes
+      providerConfig: {
+        ..._routingConfig,
+        additionalGasOverhead: NATIVE_OVERHEAD(this.chainId, amountToken, quoteToken)
```

Comment:
I'm a bit confused about what this line does, can you explain a little bit?

Is there any way in which this could mess up something?

Reply:
I think you can introduce `additionalGasOverhead` within `AlphaRouterConfig`, and that should take care of it here. Because https://github.com/Uniswap/smart-order-router/pull/395/files#diff-2fe4a092f36b8091262f7035638dc00530816382f3a1cca86927a6fd712e9c04R987-R991 made it such that top-level `ProviderConfig` always contains the `additionalGasOverhead`

Reply:
As for the context, it was probably a follow-up to https://github.com/Uniswap/smart-order-router/pull/357. Nice to have this fix, although I don't think it will actually take much effect, since we always enable tenderly simulation for gas estimate. The only scenario I can see this change can help is when there are multiple routes to quote against, and v2 route quote is slightly better than v3 route quote or mixed route quote after adjusting the gas. But this v2 route quote is actually worse, because it didn't take into account the wrap/unwrap overhead. And this probably has to be the ETH/WETH token(s) in the v2 route-through, not token in or token out, that are not in the v3/mixed route.